contracts/contracts/compensation/CompensationClaims.sol

```diff
@@ -0,0 +1,168 @@
+ *   6. The claim period starts
+ *   7. Users claim funds
+ *   8. The claim period ends
+ *   9. Governor can collect any remaing funds
+ *
+ */
+contract CompensationClaims is Governable {
+    address public adjuster;
+    address public token;
+    uint256 public end;
+    uint256 public totalClaims;
+    mapping(address => uint256) claims;
+    bool public isAdjusterLocked;
+
+    using SafeMath for uint256;
+
+    event Claim(address indexed recipient, uint256 amount);
+    event ClaimSet(address indexed recipient, uint256 amount);
+    event Start(uint256 end);
+    event Lock();
+    event Unlock();
+    event Collect(address indexed coin, uint256 amount);
+
+    constructor(address _token, address _adjuster) public onlyGovernor {
+        token = _token;
+        adjuster = _adjuster;
+        isAdjusterLocked = true;
+    }
+
+    function balanceOf(address _account) external view returns (uint256) {
+        return claims[_account];
+    }
+
+    function decimals() external view returns (uint8) {
+        return IERC20Decimals(token).decimals();
+    }
+
+    /* -- User -- */
+
+    function claim(address _recipient) external onlyInClaimPeriod nonReentrant {
+        uint256 amount = claims[_recipient];
+        require(amount > 0, "Amount must be greater than 0");
+        claims[_recipient] = 0;
+        totalClaims = totalClaims.sub(amount);
+        SafeERC20.safeTransfer(IERC20(token), _recipient, amount);
+        emit Claim(_recipient, amount);
+    }
+
+    /* -- Adjustor -- */
+
+    function setClaims(
+        address[] calldata _addresses,
+        uint256[] calldata _amounts
+    ) external notInClaimPeriod onlyUnlockedAdjuster {
+        require(
+            _addresses.length == _amounts.length,
+            "Addresses and amounts must match"
+        );
+        uint256 len = _addresses.length;
+        for (uint256 i = 0; i < len; i++) {
+            address recipient = _addresses[i];
+            uint256 newAmount = _amounts[i];
+            uint256 oldAmount = claims[recipient];
+            claims[recipient] = newAmount;
+            totalClaims = totalClaims.add(newAmount).sub(oldAmount);
+            emit ClaimSet(recipient, newAmount);
+        }
+    }
+
+    /* -- Governor -- */
+
+    function lockAdjuster() external onlyGovernor notInClaimPeriod {
+        _lockAdjuster();
+    }
+
+    function _lockAdjuster() internal {
+        isAdjusterLocked = true;
+        emit Lock();
+    }
+
+    function unlockAdjuster() external onlyGovernor notInClaimPeriod {
+        isAdjusterLocked = false;
+        emit Unlock();
+    }
+
+    function start(uint256 _seconds)
+        external
+        onlyGovernor
+        notInClaimPeriod
+        nonReentrant
+    {
+        require(totalClaims > 0, "No claims");
+        uint256 funding = IERC20(token).balanceOf(address(this));
+        require(funding >= totalClaims, "Insufficient funds for all claims");
+        _lockAdjuster();
```

Comment:
This is slightly disagreeing with the intended use sequence above, the adjuster should be locked before the claim period starts, maybe require adjuster is locked in this function or modify the intended use sequence comment.

Reply:
I guess it would be better to have a require. This makes sure you didn't forget to lock the contract, and no last minute changes to the amounts snuck in.